modules/library/main/src/main/java/org/geotools/filter/visitor/SimplifyingFilterVisitor.java

```diff
@@ -388,42 +389,68 @@ public Object visit(Id filter, Object extraData) {
         return validIdFilter;
     }
 
+    /**
+     * Return true iff there is a MatchAction for the negation of a multivalued binary comparison operation.
+     */
+    private boolean isNegatableMatchAction(MatchAction matchAction) {
+        return matchAction == MatchAction.ANY || matchAction == MatchAction.ALL;
```

Comment:
@aaime I checked every implementation of MultiValuedFilter and all either hardcode getMatchAction() to return MatchAction.ANY or set MatchAction.ANY in all constructors that do not specify it. I am confident that there are no backward compatibility concerns. It is possible to set MatchAction to null but only by using a post-MatchAction constructor, and anyone doing so should know better. Cannot happen with old code.